geps/gep-2627/index.md

```diff
@@ -0,0 +1,50 @@
+* Status: Provisional
+
+## TLDR
+
+For gateway infrastructure to be valuable we need to be able to connect clients to these gateways. A common way to achieve this is to use domain names/hostnames and DNS. Gateways define listeners that can have assigned hostnames they are also are assigned addresses and ports to express where on the network the gateway can be reached. The correct configuration of DNS is critical piece but is currently not expressible as part of Gateway API instead it is left as an exercise of the user to figure out. Bringing these two concerns closer together will serve to provide a single plane to express and view not only the proxy configuration but also the DNS based connectivity configuration and status.
+
+## Goals
+* Allow cluster operators or infrastructure providers to declaratively express which DNS service they want to use with a particular Gateway or Gateway Listener and have that DNS service interacted with via a controller to provision and configure the defined listener hostnames as DNS names and records based on the addresses assigned to the gateway instances.
+* Provide a standard CRD based API to provide an alternative to the need for "loose" APIs such as annotations that are not easily extended, standardized, versioned, or communicated about via a dedicated status. 
+* Increase portability and supportability across gateway providers and third party implementors, for this type of key configuration
+
+## Non-Goals
+
+* Cover more complex DNS routing strategies that come into play for multi-cluster topologies such as round robin, failover, health checks, weighted and geo location with this first pass. Supporting these types of use cases for distributed gateways (IE for example in different regions or multiple gateways for resilience within a region) and offering a form of global load balancing leveraging DNS is something that would be good to follow up on as a discussion point after this GEP.
+
+## Use Cases
+
+As a cluster administrator, I manage a set of domains and a set of gateways. I would like to declaratively define which DNS provider to use to configure connectivity for clients accessing these domains and my gateway so that I can see and configure which DNS provider is being used and limit which domains can be used.
+
+A a cluster administrator, I would like to have the dns names automatically populated into my specified dns zones as a set of records based on the assigned addresses of my gateways and have the status of the DNS records reported back to me, so that I do not have to undertake external automation or management of this essential task and can leverage existing kube based monitoring tools to know the status of the integration.
+
+As a cluster administrator, I would like the DNS records to be setup automatically based on the assigned gateways address and if the IP or hostname changes, I would like for the DNS to update automatically to ensure traffic continues to reach my gateway. 
```

Comment:
`This would have to be configurable, not automatic. Maybe not all hostnames will be in the same IP subnet.`  by this do you mean if a gateway is assigned multiple addresses in its spec, one public and one private for example? 

`I'm not clear on use cases where the IPaddr or hostname changes - can you give an example?`  This is really to indicate that if, the assigned address in the Gateway is changed or new listeners are added, the DNS will be updated appropriately

Reply:
```suggestion
As a cluster administrator, I would like the DNS records to be updated automatically if the `spec` of assigned gateways changes, whether those changes are for IP address or hostname. 
```

Is this a bit clearer?